modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/IndexManager.java

```diff
@@ -372,6 +372,7 @@ protected QuadTree openQuadTree() throws StoreException {
         if (!shpFiles.isLocal()) {
             return null;
         }
+        FileSystemIndexStore idxStore = null;
```

Comment:
I don't see what this change is actually accomplishing... by pushing the FileSystemIndexStore variable declaration out, I thought you were going for a cleanup attempt later, but I don't see it. 
So... what's the actual change? You need to unlock the treeURL before trying to load? That seems more of a regression, other processes could try to write on it while the read is happening....